flink-streaming-java/src/main/java/org/apache/flink/streaming/util/retryable/AsyncRetryStrategies.java

```diff
@@ -181,11 +180,12 @@ public boolean canRetry(int currentAttempts) {
         public long getBackoffTimeMillis(int currentAttempts) {
             if (currentAttempts <= 1) {
                 // equivalent to initial delay
-                return lastRetryDelay;
+                return initialDelay;
```

Comment:
I remembered that the original version was expecting to avoid `pow` evaluation by cache last value, but this was problematic due to the lack of a reset when reusing the object.
I see another way to fix is to add `reset` here：
```java
// reset to initialDelay
this.lastRetryDelay = initialDelay;
return lastRetryDelay;
```
WDYT?

Reply:
@lincoln-lil Hi lincoln, thx for comment. AFAIK, cache last value is not only problematic when reusing the object but also in situations where user will skip some attempts to call `getBackoffTimeMillis`, this will lead to inconsistent result for the same input. 

For example, User1 might use getBackoffTimeMillis(1), getBackoffTimeMillis(2), getBackoffTimeMillis(4); User2 might use getBackoffTimeMillis(1), getBackoffTimeMillis(2), getBackoffTimeMillis(3). The getBackoffTimeMillis(4) of User1 will have the same result with the getBackoffTimeMillis(3) of User2. 

WDYT?